packages/yew/src/dom_bundle/fragment.rs

```diff
@@ -0,0 +1,166 @@
+use std::collections::VecDeque;
+use std::ops::{Deref, DerefMut};
+
+use web_sys::{Element, Node};
+
+use super::BSubtree;
+use crate::html::NodeRef;
+use crate::virtual_dom::Collectable;
+
+/// A Hydration Fragment
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
+pub(crate) struct Fragment(VecDeque<Node>);
+
+impl Deref for Fragment {
+    type Target = VecDeque<Node>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for Fragment {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+impl Fragment {
```

Comment:
> I suspect this would be a lot easier to debug than running into small mismatches when the hydration mechanism changes.

There's no good way to do this other than embedding version number?
However, one could still be using a master version of yew.

The easiest way is to use a workspace and compile backend and frontend at the same time.

Reply:
> The easiest way is to use a workspace and compile backend and frontend at the same time.

Caching and "helpful" CDNs might mess with the assumption that the app is served directly. I'd rather be safe than sorry.
If you think a manually bumped version isn't enough, even a reasonable build-time dependency for getting the current git-commit hash from the repo sounds okay to me.

Reply:
> Caching and "helpful" CDNs

Usually, bundler creates a different hash each time a bundle is created. As long as you make sure that no cache is configured on your `index.html`, it will not happen.

If you didn't configure your `cache-control` policy properly, it can cause all kinds of hard to debug issues and failure to hydrate is just 1 single of them.

Reply:
> it can cause all kinds of hard to debug issues and failure to hydrate is just 1 single of them.

Yep, and I think it could be a nice reminder for people, with a relatively straight forward and easy to read message to remind them they messed up something.

Reply:
Can we do this in a future pull request?

I have a pull request planned that allows carrying information from the server-side to client side.

Reply:
As long as it lands before 0.20, if it fits better there, sure.